otelcollector/shared/configmap/mp/prometheus-config-merger.go

```diff
@@ -1311,7 +1311,13 @@ func mergeDefaultAndCustomScrapeConfigs(customPromConfig string, mergedDefaultCo
 			return
 		}
 
-		mergedConfigs := deepMerge(mergedDefaultConfigs, customPrometheusConfig)
+		var mergedConfigs map[interface{}]interface{}
+		if customPrometheusConfig["scrape_configs"] != nil {
```

Comment:
If the customer has values in the global section then this would skip it right?

For e.g. 
if the configmap is like:

```
global:
  scrape_interval: 45s
  external_label:
    label_1 : custom_value_1 
```
Maybe we should add a check to delete any section which is empty and then do the merge?

Reply:
Thanks @bragi92. FIxed

Reply:
Thanks @rashmichandrashekar, btw curious if this is the configmap provided
```
global:
  scrape_interval: 45s
  external_label:
    label_1 : custom_value_1 
  scrape_configs:
```
Is `customPrometheusConfig["scrape_configs"] == {}` in this case and it goes in the if condition or is `customPrometheusConfig["scrape_configs"] == nil` and it goes into the else condition?

Reply:
@bragi92 - Both go into the else condition.